metrics.py

```diff
@@ -427,17 +439,25 @@ def calculate_sample_statistics(self, networks_sample: np.ndarray) -> np.ndarray
         if self.requires_graph:
             networks_as_graphs = [connectivity_matrix_to_G(W, self.is_directed) for W in networks_sample]
 
+        networks_as_sparse_tensor = np_tensor_to_sparse_tensor(networks_sample)
+
         feature_idx = 0
         for metric in self.metrics:
             n_features_from_metric = metric.get_effective_feature_count(self.n_nodes)
 
             if metric.requires_graph:
                 networks = networks_as_graphs
             else:
-                networks = networks_sample
-
-            features_of_net_samples[feature_idx:feature_idx + n_features_from_metric] = metric.calculate_for_sample(
-                networks)
+                # networks = networks_sample
+                networks = networks_as_sparse_tensor
+                
+            features = metric.calculate_for_sample(networks)
```

Comment:
Again, we can get rid of all the sparse versions of `features` after converting it back to `numpy` (if there are any, I'm not sure if everything here is in place or not)